tests/parser.cpp

```diff
@@ -2872,4 +2872,23 @@ BEGIN { $x: int8 = 1; }
 )");
 }
 
+TEST(Parser, macro_expansion_error)
+{
+  BPFtrace bpftrace;
+
+  // A recursive macro expansion
+  bpftrace.macros_.emplace("M", "M+1");
+  test_parse_failure(bpftrace,
+                     "#define M M+1\n"
+                     "BEGIN { M; }",
+                     R"(
+stdin:2:9-343: ERROR: Macro recursion limit reached: M, M+1
+BEGIN { M; }
+        ~~~~
+stdin:2:9-343: ERROR: syntax error, unexpected end of file
+BEGIN { M; }
+        ~~~~
```

Comment:
I think parser inserts this, right? So not necessary to do anything fancy.
```suggestion
  test_parse_failure("#define M M+1\n"
                     "BEGIN { M; }",
                     R"(
stdin:2:9-343: ERROR: Macro recursion limit reached: M, M+1
BEGIN { M; }
        ~~~~
stdin:2:9-343: ERROR: syntax error, unexpected end of file
BEGIN { M; }
        ~~~~
```

Reply:
Actually, it doesn't :joy:. Macro parsing is done by `ClangParser` (https://github.com/bpftrace/bpftrace/pull/3824#discussion_r1974983783). I have added a comment to clarify this.

Reply:
Ah, got it. How about writing a new wrapper, maybe `test_macro_parse_failure(...)`? We recently overhauled the pass manager, so should be straightforward. Example of a I recently added: https://github.com/bpftrace/bpftrace/blob/882c1836aebb082995bda421cbfe30b43e5430aa/tests/deprecated.cpp#L11-L27

Reason is it'd be good to move the tests towards stricter input -> output and avoid reaching into classes to mock.